src/service/performance-impl.js

```diff
@@ -839,6 +879,24 @@ export class Performance {
   }
 }
 
+/**
+ * Traverse node ancestors and return the highest level amp element.
+ * Returns the given node if none are found.
+ *
+ * @param {!HTMLElement} node
+ * @return {!HTMLElement}
+ */
+function getOutermostAmpElement(node) {
+  let max = node;
+  while (node != null) {
+    node = node.parentNode;
+    if (node?.nodeName.startsWith('AMP-')) {
```

Comment:
We've already asserted `!= null`.
```suggestion
    if (node.nodeName.startsWith('AMP-')) {
```

Reply:
Wait...no we didn't. At least not about the parent!

Reply:
Reverting